lib/AST/ASTMangler.cpp

```diff
@@ -484,12 +484,20 @@ void ASTMangler::beginManglingWithAutoDiffOriginalFunction(
     appendOperator(attr->Name);
     return;
   }
+
+  auto beginManglingClangDecl = [&](const clang::NamedDecl *decl) {
+    beginManglingWithoutPrefix();
+    appendOperator(decl->getName());
+  };
+
   // For imported Clang declarations, use the Clang name in order to match how
   // DifferentiationMangler handles these.
   auto clangDecl = getClangDeclForMangling(afd);
   if (clangDecl) {
```

Comment:
Minor: to indicate that `clangDecl` does not matter after the `if` block, and make it clearer that one of the two branches will succeed, if possible, the following might be preferred:

```
if (auto clangDecl = getClangDeclForMangling(afd)) {
```